src/coreComponents/common/Path.cpp

```diff
@@ -25,12 +25,12 @@ namespace geosx
 {
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-void getAbsolutePath( std::string const & path, std::string & absolutePath )
+std::string getAbsolutePath( std::string const & path )
```

Comment:
Yes... but there's a circular inclusion problem that I didn't want to touch in this PR. Namely, `DataTypes.hpp` needs to include `Path.hpp` in order to provide some related typedefs and lambda dispatch on it. On the other hand, `Path.hpp` would need to include `DataTypes.hpp` in order to gain access to `geosx::string`.

One solution is to split a small portion of `DataTypes.hpp` into a separate header, `BasicTypes.hpp`, that will provide aliases like `localIndex`, 'globalIndex', `real64`, `string`. It can then be independently included in `Path.hpp` and `DataTypes.hpp`. I didn't do it here as it would create merge conflicts for the BoomerAMG PR, but we can address this subsequently.

Reply:
ok then

Reply:
It is intentional that we kept `std::string` here. `Path` is its own type derived exclusively from `std::string`.